spec/killbill_client/encoding_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe KillBillClient::API do
+
+  it 'should send double-encoded uri' do
+    plugin_properties = [
+        {
+            'key' => 'contractId',
+            'value' => "test"
+        },
+        {
+            'key'   => 'details',
+            'value' =>  {
+                'eventType' => 'voidEvent',
+                'transactionType' => 'void',
+                'contractType' => 'temp'
+            }.to_json
+        }
+    ].map{|hash| KVPair.from_hash(hash)}
+    options = {
+        :params => {:controlPluginName => "killbill-example-plugin"},
+        :pluginProperty => plugin_properties
+    }
+    http_adapter = DummyForHTTPAdapter.new
+    uri = http_adapter.send(:encode_params, options)
+    (uri == expected_uri).should be_true
```

Comment:
RSpec has been [upgraded](https://github.com/killbill/killbill-client-ruby/pull/22) to 3.4 in master. The syntax has changed a little bit.